Minio/ApiEndpoints/BucketOperations.cs

```diff
@@ -56,6 +56,100 @@ public async Task<ListAllMyBucketsResult> ListBucketsAsync(CancellationToken can
             return bucketList;
         }
 
+        /// <summary>
+        /// Check if a private bucket with the given name exists.
+        /// </summary>
+        /// <param name="args">BucketExistsArgs Arguments Object which has bucket identifier information - bucket name, region</param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
+        /// <returns> Task </returns>
+        /// <exception cref="InvalidBucketOperationException">When Args object or another setting is wrong</exception>
+        public async Task<bool> BucketExistsAsync(BucketExistsArgs args, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            args.Validate();
+            this.UpdateArgs(args);
+            try
+            {
+                this.restClient.Authenticator = new V4Authenticator(this.Secure, this.AccessKey, this.SecretKey, region: this.Region, sessionToken: this.SessionToken);
+                var request = args.GetRequest(this.BaseUrl, this.restClient.Authenticator);
+                await this.ExecuteTaskAsync(this.NoErrorHandlers, request, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if ( ex.GetType() == typeof(BucketNotFoundException) )
+                {
+                    return false;
+                }
+                throw;
+            }
+            return true;
+        }
+
+
+        /// <summary>
+        /// Remove the bucket with the given name.
+        /// </summary>
+        /// <param name="args">RemoveBucketArgs Arguments Object which has bucket identifier information like bucket name .etc.</param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
+        /// <returns> Task </returns>
+        /// <exception cref="InvalidBucketOperationException">When Args object or another setting is wrong</exception>
+        public async Task RemoveBucketAsync(RemoveBucketArgs args, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            args.Validate();
+            this.restClient.Authenticator = new V4Authenticator(this.Secure, this.AccessKey, this.SecretKey, region: this.Region, sessionToken: this.SessionToken);
+            var request = args.GetRequest(this.BaseUrl, this.restClient.Authenticator);
+            var bktRegion = await GetRegion(args.BucketName).ConfigureAwait(false);
+            Uri requestUrl = RequestUtil.MakeTargetURL(this.BaseUrl, this.Secure, args.BucketName, region: bktRegion , usePathStyle: false);
```

Comment:
this entire code block - lines 98 thru 101 should all be factored into CreateRequest. It is repeated across API calls. CreateRequest can take Args as a parameter, and cast to specific type for GetRequest

Reply:
Yes, I'll do that change.